TASVideos.Core/Services/ForumService.cs

```diff
@@ -97,11 +107,44 @@ public void CacheLatestPost(int forumId, int topicId, LatestPost post)
 		}
 	}
 
-	public void ClearCache()
+	public void CacheNewPostActivity(int forumId, int topicId, DateTime CreateTimestamp)
+	{
+		if (_cacheService.TryGetValue(TopicActivityCacheKey, out Dictionary<int, Dictionary<int, DateTime>> forumActivity))
+		{
+			forumActivity.TryGetValue(forumId, out Dictionary<int, DateTime>? subforumActivity);
+			if (subforumActivity == null)
+			{
+				subforumActivity = new Dictionary<int, DateTime>();
+			}
+			subforumActivity[topicId] = CreateTimestamp;
+			forumActivity[forumId] = subforumActivity;
+			_cacheService.Set(TopicActivityCacheKey, forumActivity, Durations.OneDayInSeconds);
+		}
+		if (_cacheService.TryGetValue(TopicActivityJsonCacheKey, out Dictionary<int, string> jsonForumActivity))
+		{
+			var dictSubforumActivity = new Dictionary<int, string>();
+			jsonForumActivity.TryGetValue(forumId, out string? jsonSubforumActivity);
+			if (jsonSubforumActivity != null)
+			{
+				dictSubforumActivity = JsonSerializer.Deserialize<Dictionary<int, string>>(jsonSubforumActivity) ?? new Dictionary<int, string>();
+			}
+			dictSubforumActivity[topicId] = CreateTimestamp.UnixTimestamp().ToString();
+			jsonForumActivity[forumId] = JsonSerializer.Serialize(dictSubforumActivity);
```

Comment:
Is there any reason you are serializing before caching? You get that for free from Redis Cache, and for memory cache it can just be the native class

Reply:
I'm not serializing for the cache and later deserializing. I *need the json itself* to embed the data into html and send it javascript.